salt/transport/tcp.py

```diff
@@ -432,7 +432,15 @@ def socket(self):
 
     def close(self):
         if self._socket is not None:
-            self._socket.shutdown(socket.SHUT_RDWR)
+            try:
+                self._socket.shutdown(socket.SHUT_RDWR)
+            except socket.error as exc:
+                if exc.errno == 107:
```

Comment:
I think it would be cleaner if you added a:

```
import errno
```

to the top and then made this:

```
if exc.errno == errno.ENOTCONN:
```